### ISSUE TITLE
API not resettingvst 00:00 UTC

### ISSUE THREAD

**Issue by hwb45** [oziee/ha-solcast-solar#291]
### The problem

I see from the release notes that you jiw reset the API count at 00:00 UTC, but my system just keeps incrementing the count. Do I need yo anything special at 00:00?

### What version of Solcast Integration has the issue?

v4.0.20

### What version of Home Assistant are you using?

2024.5.3

### Logs debug/info/error output

```txt
This error originated from a custom integration.

Logger: custom_components.solcast_solar.solcastapi
Source: custom_components/solcast_solar/solcastapi.py:689
integration: Solcast PV Forecast (documentation, issues)
First occurred: 15 May 2024 at 23:00:33 (1 occurrences)
Last logged: 15 May 2024 at 23:00:33

SOLCAST - Exceeded Solcast API allowed polling limit
```

**Comment by changeofaspect** [comment]
Solcast: v4.0.20, Home Assistant version: 2024.5.3
I am having the same issue, the API count carries over to the next day. If I log into my Solcast toolkit, I can see that I have used 3 out of my limit of 50 but the integration says that I have used 19. 

If I 'reload' the integration it corrects the issue and shows the correct count of 3. 
This started yesterday.

The Warning that is logged says that I have exceeded my API allowed polling limit. However, as an original user, I have a limit of 50 and not 10 (for newer accounts). The integration thinks that I have used 19 which does not exceed the 50 that I am allowed.
